libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c

```diff
@@ -952,36 +952,26 @@ static OTA_Err_t prvRequestDataHandler( OTA_EventData_t * pxEventData )
         /* Request data blocks. */
         xErr = xOTA_DataInterface.prvRequestFileBlock( &xOTA_Agent );
```

Comment:
Should this be within `if( xOTA_Agent.ulRequestMomentum < otaconfigMAX_NUM_REQUEST_MOMENTUM )` ?
Otherwise we sent request for the last file block and still mark it as failed before the response for the last retry.

Reply:
Make sense, fixed.